### PR TITLE
Support for the FICO Xpress solver

### PR DESCRIPTION
**Pull request by merraksh**
## Changes proposed in this Pull Request

Provide support fo the FICO Xpress commercial solver, in a way similar to the support given by the other two commercial solvers (Cplex, Gurobi) via a problem file to be read and solved.

## Checklist

- Documented new function `run_and_read_xpress`;
- A note for the release notes `doc/release_notes.rst` of the upcoming release is included.

## Acknowledgments
Acknowledgment to the team at Transnet for their significant contribution to this Xpress-PyPSA binding.

### CODE REVIEW

File: doc/release_notes.rst [hunk 1]
```diff
@@ -7,6 +7,8 @@ Upcoming Release
 
 .. warning:: The features listed below are not released yet, but will be part of the next release! To use the features already you have to install the ``master`` branch, e.g. ``pip install git+https://github.com/pypsa/pypsa#egg=pypsa``.
 
+* Added support for the FICO Xpress commercial solver.
```
Comment (FabianHofmann) [inline doc/release_notes.rst:10]:
```suggestion
* Added support for the FICO Xpress commercial solver for `pyomo=False`
```

File: pypsa/linopt.py [hunk 2]
```diff
@@ -803,3 +803,74 @@ def run_and_read_gurobi(n, problem_fn, solution_fn, solver_logfile,
...
+    if termination_condition not in ["optimal"]:
+        return status, termination_condition, None, None, None
+
+    var = [str(v) for v in m.getVariable()]
+    variables_sol = pd.Series(m.getSolution(var), index=var).pipe(set_int_index)
+
+    try:
+        dual = [str(d) for d in m.getConstraint()]
+        constraints_dual = pd.Series(m.getDual(dual), index=dual).pipe(set_int_index)
+    except:
```
Comment (FabianHofmann) [inline pypsa/linopt.py:867]: Could you add the error type which is raise here?

Comment (merraksh) [reply pypsa/linopt.py]: Good point, just pushed. It should catch xpress.SolverError when getDual() fails (due to e.g. unsuccessful solve). All other types of errors (InterfaceError, ModelError) are not caught, but are much more unlikely: m.getConstraint() always returns a list, possibly empty.

**Comment by FabianHofmann** [comment]
Great! thanks for the contribution!

**Comment by merraksh** [comment]
Wonderful, thanks for the quick response.

**Pull request merged**